src/model/Space.cpp

```diff
@@ -918,6 +924,121 @@ namespace model {
       return result;
     }
 
+    std::vector<Surface> Space_Impl::findSurfacesWithIncorrectOrientationRaycasting() const {
```

Comment:
This was my original method to detect incorrect orientations. But it doesn't work for non convex spaces. This is unused at the moment, but not exposed publicly. I am not remove the code because there might be other applications (calculating viewFactors for eg).

Anyways the idea was to take the surface's outwardNormal, reverse it so it points inward, then shoot a ray in that direction from the centroid of the surface. You then check if any other surface's plane receives the ray. if you can't find any, the outward normal is reversed.